src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java

```diff
@@ -51,37 +45,27 @@ public ParallelFlowExecutorServiceImpl(ExecutorService threadPoolExecutor,
      */
     @Override
     public void execute() {
-        while (FLAG) {
-            threadPoolExecutor.execute(tasksFactory.createExecutionWorker(
-                    service, scenarioHandler.getScenario(), proxyHandler.getProxy()));
+
+        threadPoolExecutor.execute(this::runParallelExecute);
```

Comment:
1. With this approach, you use one thread from **threadPoolExecutornot** not effectively (not to complete the task from Queue), but you use it for this:
![image](https://github.com/GFLCourses5/worker/assets/140727960/e8c52aa6-4d1f-4d9b-a76a-04bc65d838e8).

Reply:
Give me some more details why it isn't efficient? And cany you also explain why do you have to use run of new thread in main thread?